src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -93,9 +93,10 @@ public FeatureManagementKeyValueAdapter(FeatureFilterTracing featureFilterTracin
 
                     keyValues.Add(new KeyValuePair<string, string>($"{variantsPath}:{FeatureManagementConstants.Name}", featureVariant.Name));
 
-                    if (featureVariant.ConfigurationValue != null)
+                    foreach (KeyValuePair<string, string> kvp in new JsonFlattener().FlattenJson(featureVariant.ConfigurationValue))
```

Comment:
I assume this works with `null`? We'll just actually get a key value pair with null as the value?

Reply:
At the moment it would have no key value pairs and just skip the loop. If you evaluate it with feature management `Variant.Configuration` would just be null either if the key exists as null or is missing entirely. It seems like having the key present set to null would be clearer though?

Reply:
The way we handle json key values in the provider currently is to just exclude the key though, so could also be a question of keeping that consistent.

Reply:
Kidding - it does just add the key with null value

Reply:
I guess `null` could mean a lot of things. The ones I was thinking about are:

null value
```json
"variants": [
  "name": "test",
  "configuration_value": null
]
```
Should have a kv pair "configuration_value" => null

Missing config_value
```json
"variants": [
  "name": "test",
  "configuration_reference": "SomethingWeHaventDoneYet"
]
```
Should not have a kv pair "configuration_value" => null

Reply:
I think that makes sense, would just need to alter the first scenario since that's where it returns empty string now. This scenario is a bit different from how we use `FlattenJson` for filter parameters since portal doesn't allow parameters to be null or omitted, so it should be ok to change that.

Reply:
Yea, I can change the first scenario to return null instead of empty string. That would mean they get treated as effectively the same, where trying to get the value for that key will return null.